LibAddonMenu-2.0/exampleoptions.lua

```diff
@@ -9,109 +9,118 @@ local panelData = {
-    },
-    [3] = {
-        type = "dropdown",
-        name = "My Dropdown",
-        tooltip = "Dropdown's tooltip text.",
-        choices = {"table", "of", "choices"},
-        getFunc = function() return "of" end,
-        setFunc = function(var) print(var) end,
-        width = "half",	--or "half" (optional)
-        warning = "Will need to reload the UI.",	--(optional)
-    },
-    [4] = {
-        type = "dropdown",
-        name = "My Dropdown",
-        tooltip = "Dropdown's tooltip text.",
-        choices = {"table", "of", "choices"},
-        getFunc = function() return "of" end,
-        setFunc = function(var) print(var) end,
-        width = "half",	--or "half" (optional)
-        warning = "Will need to reload the UI.",	--(optional)
-    },
-    [5] = {
-        type = "slider",
-        name = "My Slider",
-        tooltip = "Slider's tooltip text.",
-        min = 0,
-        max = 20,
-        step = 1,	--(optional)
-        getFunc = function() return 3 end,
-        setFunc = function(value) d(value) end,
-        width = "half",	--or "half" (optional)
-        default = 5,	--(optional)
-    },
-    [6] = {
-        type = "button",
-        name = "My Button",
-        tooltip = "Button's tooltip text.",
-        func = function() d("button pressed!") end,
-        width = "half",	--or "half" (optional)
-        warning = "Will need to reload the UI.",	--(optional)
-    },
-    [7] = {
-        type = "submenu",
-        name = "Submenu Title",
-        tooltip = "My submenu tooltip",	--(optional)
-        controls = {
-            [1] = {
-                type = "checkbox",
-                name = "My Checkbox",
-                tooltip = "Checkbox's tooltip text.",
-                getFunc = function() return true end,
-                setFunc = function(value) d(value) end,
-                width = "half",	--or "half" (optional)
-                warning = "Will need to reload the UI.",	--(optional)
-            },
-            [2] = {
-                type = "colorpicker",
-                name = "My Color Picker",
-                tooltip = "Color Picker's tooltip text.",
-                getFunc = function() return 1, 0, 0, 1 end,	--(alpha is optional)
-                setFunc = function(r,g,b,a) print(r, g, b, a) end,	--(alpha is optional)
-                width = "half",	--or "half" (optional)
-                warning = "warning text",
-            },
-            [3] = {
-                type = "editbox",
-                name = "My Editbox",
-                tooltip = "Editbox's tooltip text.",
-                getFunc = function() return "this is some text" end,
-                setFunc = function(text) print(text) end,
-                isMultiline = false,	--boolean
-                width = "half",	--or "half" (optional)
-                warning = "Will need to reload the UI.",	--(optional)
-                default = "",	--(optional)
-            },
+local optionsTable = {}
+

```

Comment:
It's a good idea to make it easier to move around the options, but I feel that splitting it up into several insert calls is not the way to go. Instead I think it would be better to keep it in one table, but drop the index like so:
```lua
local optionsTable = {
    {
        type = "header",
        ...
    },
    {
        type = "description",
        ...
    },
    ...
}
```